### DISCUSSION TITLE
Will this project accept a PR to support nvim-dap with coc.nvim?

### DISCUSSION THREAD

**Discussion by Frederick888** [mfussenegger/nvim-jdtls#450] [Ideas]
I'm still using coc.nvim (yeah, I know, I know 🤦, but it works well for me), which only has out-of-the box support for vimspector via [coc-java-debug](https://github.com/dansomething/coc-java-debug).

But since I already migrated to nvim-dap/-ui, I started writing my own nvim-dap configs using [jdtls/dap.lua](https://github.com/mfussenegger/nvim-jdtls/blob/master/lua/jdtls/dap.lua) as a reference.

Soon I realised that I was basically copying most of the stuff from this file while only changing the way it makes LSP calls to `vim.fn['CocRequest']('java', 'workspace/executeCommand', ...)`.

So I'm curious if nvim-jdtls is open to a PR to support nvim-dap with coc.nvim (and add an option to skip setting up jdt.ls itself), or it's a non-goal and I'm better off forking the project for this purpose? Thank you.

**Comment by Frederick888** [comment]
This setup seems to be working:

- nvim-jdtls changes: https://github.com/Frederick888/nvim-jdtls/commits/coc-nvim-dap
- coc-java changes: https://github.com/Frederick888/coc-java/commits/user-bundles
- coc config
```json
{
  "java.jdt.ls.bundles": [
    "/home/frederick/Programming/Java/java-debug/com.microsoft.java.debug.plugin/target/com.microsoft.java.debug.plugin-0.44.0.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/com.microsoft.java.test.plugin-0.38.1.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-jupiter-api_5.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-jupiter-engine_5.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-jupiter-migrationsupport_5.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-jupiter-params_5.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-commons_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-engine_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-launcher_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-runner_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-suite-api_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-suite-commons_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-platform-suite-engine_1.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/junit-vintage-engine_5.9.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/org.apiguardian.api_1.1.2.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/org.eclipse.jdt.junit4.runtime_1.3.0.v20220609-1843.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/org.eclipse.jdt.junit5.runtime_1.1.100.v20220907-0450.jar",
    "/home/frederick/Programming/JavaScript/vscode-java-test/server/org.opentest4j_1.2.0.jar"
  ],
```

It doesn't look possible to access server capabilities from vimscript/lua so I just hard coded `vscode.java.test.findTestTypesAndMethods` for coc though.

**Comment by mfussenegger** [comment]
> So I'm curious if nvim-jdtls is open to a PR to support nvim-dap with coc.nvim (and add an option to skip setting up jdt.ls itself), or it's a non-goal and I'm better off forking the project for this purpose?

That's a non-goal I don't use coc.nvim myself and don't want to have to consider alternative clients when making changes. I'm afraid you'll have to fork the project.

**Reply by Frederick888** [reply]
Thanks for the quick response!

I actually implemented a super hacky way for this to work without forking nvim-jdtls: https://github.com/Frederick888/coc-lsp-adapter.nvim. Fingers crossed that it'll be easier to maintain than forking. In the long run I'll have to come up with something else I guess.
